openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -501,24 +501,27 @@ def _verify_user_and_log_mismatch(request, response, userid_in_session):
 
             try:
 
-                if LOG_REQUEST_USER_CHANGE_HEADERS.is_enabled() and request.user.id is not None:
+                if LOG_REQUEST_USER_CHANGE_HEADERS.is_enabled():
 
                     # add a session hash custom attribute for all requests to help monitoring
-                    #   requests that come both before and after a mistmatch
+                    #   requests that come both before and after a mismatch
                     if hasattr(request, 'cookie_session_field'):
                         session_hash = obscure_token(request.cookie_session_field)
                         set_custom_attribute('safe_sessions.session_id_hash.parsed_cookie', session_hash)
 
-                    # log request header if this user id was involved in an earlier mismatch
-                    log_request_headers = cache.get(
-                        SafeSessionMiddleware._get_recent_user_change_cache_key(request.user.id), False
-                    )
-                    if log_request_headers:
-                        log.info(
-                            f'SafeCookieData request header for {request.user.id}: '
-                            f'{SafeSessionMiddleware._get_encrypted_request_headers(request)}'
+                    user_id = userid_in_session or hasattr(request, 'user') and request.user.id
```

Comment:
**Reviewer Note**: I added this in the off chance that either `userid_in_session` or `request.user.id` could be `None` while the other contains the actual user id. I was going to add additional custom attributes to learn if this could ever happen, but pylint started complaining about the method having too many statements. Decided on this for expediency.

Reply:
Probably worth leaving a comment to that effect.

Reply:
Done.